content/en/tracing/trace_collection/custom_instrumentation/java/otel.md

```diff
@@ -17,118 +17,201 @@ further_reading:
-ContextKey<OtelSpan> localRootSpanKey = ContextKey.named("opentelemetry-traces-local-root-span");
-Span rootSpan = Context.current().get(localRootSpanKey);
-rootSpan.setAttributes("some-key", "some-value");
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+import io.opentelemetry.semconv.ResourceAttributes;
+import java.util.concurrent.TimeUnit;
+
+public class Example {
+
+  public void begin() {
+    tracer = getOpenTelemetry().getTracer("my-scope", "0.1.0");
+    Span parentSpan = tracer.spanBuilder("begin").startSpan();
+    try (Scope scope = parentSpan.makeCurrent()) {
+      createChildSpan();
+    } finally {
+      parentSpan.end();
+    }
+  }
+
+  private void createChildSpan() {
+    Span childSpan = tracer.spanBuilder("child-span").startSpan();
+    try {
+      Span rootSpan = Context.current()
+        .get(ContextKey.named("opentelemetry-traces-local-root-span"));
+      rootSpan.setAttribute("my-attribute", "my-attribute-value");
+      rootSpan.setStatus(StatusCode.ERROR, "Some error details...");
+    } finally {
+      childSpan.end();
+    }
+  }
+
+  public OpenTelemetry getOpenTelemetry() {
+    Resource resource = Resource.getDefault().toBuilder()
+      .put(ResourceAttributes.SERVICE_NAME, "my-service").build();
+    OtlpGrpcSpanExporter otlpGrpcSpanExporter = OtlpGrpcSpanExporter.builder()
+      .setTimeout(5, TimeUnit.SECONDS).build();
+    SdkTracerProvider setTracerProvider = SdkTracerProvider.builder()
+      .addSpanProcessor(BatchSpanProcessor.builder(otlpGrpcSpanExporter)
+        .setScheduleDelay(100, TimeUnit.MILLISECONDS).build())
+      .setResource(resource)
+      .build();
+    return OpenTelemetrySdk.builder().setTracerProvider(setTracerProvider)
+      .buildAndRegisterGlobal();
+  }
+}
 ```
 
-**Note:** If there isn't a current or local root span, the returned span is invalid, not `null`, and attributes are not set.
+## Adding spans
+
+If you aren't using a [supported framework instrumentation][17], or you would like additional depth in your application's [traces][16], you may want to add custom instrumentation to your code for complete flame graphs or to measure execution times for pieces of code.
+
+If modifying application code is not possible, use the environment variable dd.trace.methods to detail these methods.
 
-### Add custom spans using annotations
+If you have existing @Trace or similar annotations, or prefer to use annotations to complete any incomplete traces within Datadog, use Trace Annotations.
```

Comment:
I suggest replacing that sentence with: 

_Traces may also be created using the OTel `@WithSpan` annotation as described in the following section._

Reply:
@am312 - Thanks, good note. I replaced the sentence with your suggestion. 👍